src/rgw/rgw_rados.cc

```diff
@@ -1894,6 +1894,14 @@ int RGWRados::Bucket::List::list_objects_ordered(
 	next_marker = index_key;
       }
 
+      if (cur_marker.instance.empty()) {
+        if (cur_marker.name == index_key.name) continue;
```

Comment:
At least when there is no instance in the marker (i.e., instance is empty), the listing should start after the marker. Do you have an example where that's not the case?

In the case where there is an instance in the marker, again I would expect the listing to start *after* the name/instance. Do you have any output where that's not the case?